src/puya/ir/arc4_router.py

```diff
@@ -561,11 +561,8 @@ def map_param_wtype_to_arc4_tuple_type(wtype: wtypes.WType) -> wtypes.WType:
             [map_param_wtype_to_arc4_tuple_type(a.wtype) for a in non_transaction_args[14:]]
         )
         last_arg = load_abi_arg(15, args_overflow_wtype, location)
-        decoded_args = create_temporary_assignment(
-            value=arc4_decode(
-                last_arg, wtypes.WTuple.from_types(args_overflow_wtype.types), location
-            ),
-            location=location,
+        decoded_args = arc4_decode(
```

Comment:
It does get cleaned up by the optimizer but the change here to not use a temporary variable creates some horrendous (unoptimized) teal. Perhaps we should swap to doing indexed access instead of decode => tuple item access if you're not a fan of the tempvar

Reply:
I'm not too worried about the unoptimised TEAL here, especially for a niche case like >15 ABI args.

Reply:
Oh btw, it's not just a case of "it gets cleaned up by the optimizer", it makes a huge positive difference to output when there's >15 args.

Reply:
https://github.com/algorandfoundation/puya/pull/80/commits/1fb40133ce2efd28b8c224a48abe29c6fad26a02#diff-32d307c813cea76e077972c004db204cc8c3a9387ffd7cd35cc822e0c9cb7f07

Reply:
> it makes a huge positive difference to output
Only because our stack allocation still sucks :P

Reply:
Pushed a [commit](https://github.com/algorandfoundation/puya/pull/80/commits/d2cf74f4b3559cae6c563420bf82ced8ffb1bdf8) to swap to indexed access. No change to the optimised output but a big change to the raw IR which should mean less work for the later stages of the compiler to do

Reply:
@tristanmenzel  nice, awesome work